cypress/e2e/models/migration/migration-waves/migration-wave.ts

```diff
@@ -277,4 +276,44 @@ export class MigrationWave {
         cy.contains("li", date.toLocaleString("en-us", { month: "long" })).click();
         cy.contains("button:not([disabled])", `${date.getDate()}`).first().click();
     }
+
+    public clickOnApplicationStatus() {
+        cy.contains(this.name)
+            .parents("tr")
+            .within(() => {
+                cy.get(MigrationWaveView.applicationStatusColumn).click();
+            });
+    }
+
+    public removeApplication(applicationName) {
+        cy.contains(this.name)
+            .parent()
+            .parent()
+            .should(($element) => {
+                expect(
+                    $element,
+                    `Migration Wave with name ${this.name} is not expanded`
+                ).to.have.class(MigrationWaveView.waveExpanded);
+            });
+
+        cy.contains(applicationName)
+            .parent()
+            .within(() => {
+                cy.get(MigrationWaveView.removeApplicationButton).click();
+            });
+    }
+
+    public createTracker() {
+        cy.contains(this.name)
+            .parent()
+            .parent()
+            .should(($element) => {
+                expect(
+                    $element,
+                    `Migration Wave with name ${this.name} is not expanded`
```

Comment:
We can put this code in a method called 
isExpandable()

and then methods like removeApplication or createTracker() can just check if the wave isExpandable() = true , then do the rest of the work .
else return .